src/NodeVisitor/YieldNotReadyNodeVisitor.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\NodeVisitor;
+
+use Twig\Environment;
+use Twig\Node\Expression\AbstractExpression;
+use Twig\Node\Node;
+use Twig\Node\YieldReady;
+
+/**
+ * @internal to be removed in Twig 4
+ */
+final class YieldNotReadyNodeVisitor implements NodeVisitorInterface
+{
+    private $useYield;
+    private $yieldReadyNodes;
+
+    public function __construct(bool $useYield)
+    {
+        $this->useYield = $useYield;
+    }
+
+    public function enterNode(Node $node, Environment $env): Node
+    {
+        $class = \get_class($node);
+
+        if ($node instanceof AbstractExpression || isset($this->yieldReadyNodes[$class])) {
+            return $node;
+        }
+
+        if (!$this->yieldReadyNodes[$class] = (new \ReflectionClass($class))->getAttributes(YieldReady::class)) {
```

Comment:
should we really store the array of ReflectionAttribute instances there ? I suggest storing only a boolean in the array.